tests/Doctrine/Tests/ODM/PHPCR/Query/Builder/QueryBuilderTest.php

```diff
@@ -100,4 +100,20 @@ public function testPrimaryAlias()
         $this->node->fromDocument('Foobar', 'f');
         $this->assertEquals('f', $this->node->getPrimaryAlias());
     }
+
+    public function testCloningQueryBuilderUniqueConverterInstance()
+    {
+        $reflection = new \ReflectionClass(get_class($this->node));
+        $property = $reflection->getProperty('converter');
+        $property->setAccessible(true);
+
+        $this->node->setConverter($this->getMock('Doctrine\ODM\PHPCR\Query\Builder\ConverterInterface'));
+
+        $clone = clone $this->node;
+
+        $hash = spl_object_hash($property->getValue($this->node));
+        $hashClone = spl_object_hash($property->getValue($clone));
+
+        $this->assertNotEquals($hash, $hashClone, 'Cloned instance of QueryBuilder does not have an unique Converter instance.');
```

Comment:
You could simply do `$this->assertNotSame($property->getValue($this->node), $property->getValue($clone))`.